File: NineChapters/02.BinarySearch_SortedArray/Python_code/remove_duplicates_from_sorted_array.py
class Solution:
    """
    @param A: a list of integers
    @return an integer
    """
    def removeDuplicates(self, A):
        if A is None or len(A) <= 1:
            return len(A)
        size = 0
        for i in xrange(1, len(A)):
            if A[i] != A[size]:
                size += 1
                A[size] = A[i]
        A[size + 1:] = []
        return size + 1

File: NineChapters/02.BinarySearch_SortedArray/Java_Code/RemoveDuplicatesFromSortedArray.java
public class Solution {
    /**
     * @param A: a array of integers
     * @return : return an integer
     */
    public int removeDuplicates(int[] nums) {
        int length = nums.length;
        if (length <= 1) {
            return length;
        }
        int p = 0;
        for (int i = 1; i < length; i++) {
            if (nums[i] != nums[p]) {
                nums[++p] = nums[i];
            }
        }
        return p + 1;
    }
}